src/Mercury.test.ts

```diff
@@ -0,0 +1,85 @@
+import BaseTest, { test, assert } from '@sprucelabs/test'
+import MercuryMock from './MercuryMock'
+import { IMercuryEventContract } from './types/mercuryEvents'
+import { MercurySubscriptionScope } from './types/subscriptions'
+
+export const SpruceEvents = {
+	core: {
+		DidEnter: 'did-enter',
+		DidLeave: 'did-leave'
+	},
+	booking: {
+		GetAppointments: 'booking:get-appointments'
+	}
+} as const
+
+interface IMyEventContract extends IMercuryEventContract {
+	'did-enter': {
+		emitPayload: {
+			userId: string
+			enteredAt: string
+		}
+		responsePayload: {
+			somethingElse: string
+		}
+	}
+	'did-leave': {
+		responsePayload: {
+			userId: string
+		}
+	}
+	'booking:get-appointments': {
+		responsePayload: {
+			appointmentIds?: string[]
+		}
+	}
+}
+
+export default class MercuryTest extends BaseTest {
+	private static mercury: MercuryMock<IMyEventContract>
+
+	protected static async beforeAll() {
+		this.mercury = new MercuryMock({
+			spruceApiUrl: 'https://local-api.spruce.ai'
+		})
+	}
+
+	@test('Can create a mercury instance')
+	protected static async createMercury() {
+		assert.isOk(this.mercury)
+	}
+
+	@test('Can create an event contract')
+	protected static async createEventContract() {
+		this.mercury.setMockResponse({
+			eventName: SpruceEvents.core.DidEnter,
+			payload: {
+				somethingElse: 'blah'
+			}
+		})
+
+		let numCallbacks = 0
+
+		this.mercury.on(
+			{
+				eventName: SpruceEvents.core.DidEnter,
+				scope: MercurySubscriptionScope.AnonymousGlobal
+			},
+			options => {
+				console.log(options.payload.somethingElse)
+				numCallbacks += 1
+			}
+		)
+
+		const result = await this.mercury.emit({
+			eventName: 'did-enter',
+			payload: {
+				userId: 'asdf',
```

Comment:
given the name of this test, should it be testing the payload matches the type in some way? Sometimes I even put `(will always pass, types will fail)` in some of my test descriptions, but I think there is more going on here.

You may want to pull payload out and do some `assert.expectType` checks.

Type checking is hard

Reply:
The checking is below where we do: assert.isOk(result.responses[0].payload.somethingElse)

If there's a linting error there it will fail...for example just adding this line will make the test fail:
`console.log(result.responses[0].payload.notreal)`

I'll add a little better check: